### PR TITLE
usability: links to methods and configs [risk: low]

### PR DESCRIPTION
**Pull request by davidangb**
DataBiosphere/firecloud-app#111

For usability, this PR adds linkifies a few parts of the UI:
1. When viewing a single submission, or a single workflow inside a submission, link to the workspace's method config on which the submission ran.
2. When viewing a single method config inside a workspace, link to the referenced Agora method, if it has not been redacted. (no links for dockstore)
3. When viewing the list of method configs inside a workspace, link to their referenced Agora methods, if they have not been redacted. (no links for dockstore)

I chose to NOT add links from the workspace's list of submissions back to each referenced config. To do this right, I'd have to make an extra ajax call to get all known configs, in order to handle configs that have been updated/deleted since the submission ran. This would be an extra performance hit. If this feature is really desired, I'd do it in a separate PR.

**Workspace method config detail** (snapshot id value is a link):
config-detail

**Workspace method config list:**
config-list

**Submission detail, config unavailable:**
submission-detail-deleted

**Submission detail, config reachable:**
submission-detail-active

---

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [x] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by davidangb** [comment]
first automation run had two failures:
```
[info] Run completed in 26 minutes, 43 seconds.
[info] Total number of tests run: 123
[info] Suites: completed 27, aborted 0
[info] Tests: succeeded 121, failed 2, canceled 0, ignored 7, pending 0
[info] *** 2 TESTS FAILED ***
[error] Failed tests:
[error] 	org.broadinstitute.dsde.firecloud.test.security.AuthDomainMultiGroupsSpec
[error] 	org.broadinstitute.dsde.firecloud.test.metadata.DataDownloadSpec
[error] (Test / testOnly) sbt.TestsFailedException: Tests unsuccessful
[error] Total time: 1656 s, completed Nov 2, 2018 6:09:24 PM
```

* `DataDownloadSpec` passed in https://fc-jenkins.dsp-techops.broadinstitute.org/job/swatomation-pipeline/12314/
* `AuthDomainMultiGroupsSpec` passed in https://fc-jenkins.dsp-techops.broadinstitute.org/job/swatomation-pipeline/12313/

File: src/cljs/main/broadfcui/page/workspace/workspace_common.cljs [hunk 1]
```diff
@@ -47,6 +48,23 @@
 (defn config->id [config]
   (select-keys config [:namespace :name]))
 
+(defn- referenced-method-parts [config]
+  ((juxt :sourceRepo :methodNamespace :methodName :methodPath :methodVersion) (:methodRepoMethod config)))
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/page/workspace/workspace_common.cljs]: This strikes me as a little weird. Maybe:
```clojure
(replace (:methodRepoMethod config) [:sourceRepo :methodNamespace :methodName :methodPath :methodVersion])
```

File: src/cljs/main/broadfcui/page/workspace/workspace_common.cljs [hunk 2]
```diff
@@ -47,6 +48,23 @@
...
+
+(defn- method-as-text [config]
+  (clojure.string/join "/" (referenced-method-parts config)))
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/page/workspace/workspace_common.cljs]: nit: import `clojure.string :as string` and then `string/join`

**Review by MattPutnam — APPROVED** [review]

File: src/cljs/main/broadfcui/page/workspace/workspace_common.cljs (hunk 1 above)
Comment (davidangb) [reply src/cljs/main/broadfcui/page/workspace/workspace_common.cljs]: thanks for this - the `juxt` was cut-paste from the previous table definition and I also found it weird but didn't investigate anything better - I like your suggestion a lot more!

File: src/cljs/main/broadfcui/components/entity_details.cljs [hunk 3]
```diff
@@ -78,7 +79,12 @@
             "agora" (list
                      (make-field :namespace "Namespace")
                      (make-field :name "Name")
-                     (make-field :snapshotId "Snapshot ID" :dropdown? true))
+                     (make-field :snapshotId "Snapshot ID" :dropdown? true
+                                 :render (fn [snapshotId]
```
Comment (zarsky-broad) [inline src/cljs/main/broadfcui/components/entity_details.cljs]: Kebab case the param

File: src/cljs/main/broadfcui/page/workspace/monitor/submission_details.cljs [hunk 4]
```diff
@@ -186,7 +186,20 @@
              [:span {:style {:fontWeight 500}} (:methodConfigurationNamespace submission)]]
             [:div {}
              [:div {:style {:fontWeight 200 :display "inline-block" :width 90}} "Name:"]
-             [:span {:style {:fontWeight 500}} (:methodConfigurationName submission)]])
+             ;; if we were able to retrieve the method config (see :load-details), then display
+             ;; the config's name as a link. Else, display it as text with an informative tooltip.
+             (if (:ws-config @state)
```
Comment (zarsky-broad) [inline src/cljs/main/broadfcui/page/workspace/monitor/submission_details.cljs]: Destructure state and props

Comment (zarsky-broad) [reply src/cljs/main/broadfcui/page/workspace/monitor/submission_details.cljs]: Also wrap this in a let and destructure the things you're using from `submission`

File: src/cljs/main/broadfcui/page/workspace/workspace_common.cljs [hunk 5]
```diff
@@ -47,6 +48,23 @@
...
   (select-keys config [:namespace :name]))
 
+(defn- referenced-method-parts [config]
+  (replace (:methodRepoMethod config) [:sourceRepo :methodNamespace :methodName :methodPath :methodVersion]))
+
+(defn- method-as-text [config]
+  (string/join "/" (referenced-method-parts config)))
+
+(defn- method-as-maybe-link [config]
+  (let [redacted? (:redacted? config)
```
Comment (zarsky-broad) [inline src/cljs/main/broadfcui/page/workspace/workspace_common.cljs]: Use :keys to destructure multiple at once

File: src/cljs/main/broadfcui/page/workspace/workspace_common.cljs [hunk 6]
```diff
@@ -74,9 +92,10 @@
                                       (if (= repo "dockstore") "Dockstore" "FireCloud"))
                       :render (fn [repo] [:span {:style {:fontWeight 200}} repo])}
                      {:header "Method" :initial-width 800
-                      :column-data (comp (juxt :sourceRepo :methodNamespace :methodName :methodPath :methodVersion) :methodRepoMethod)
-                      :as-text (partial clojure.string/join "/")
-                      :render (partial apply style/render-entity)}]}
+                      :sort-by :text
+                      :as-text method-as-text
+                      :render method-as-maybe-link
+                     }]}
```
Comment (zarsky-broad) [inline src/cljs/main/broadfcui/page/workspace/workspace_common.cljs]: Move these up a line

**Review by zarsky-broad — APPROVED** [review]

**Pull request merged**